WordPress/Sniffs/PHP/EnqueuedCheckSniff.php

```diff
@@ -0,0 +1,128 @@
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	protected $group_name = 'Enqueued';
+
+	/**
+	 * List of enqueued functions that need to be check to make sure
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_style/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_style/
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array <string function_name> => <bool always needed ?>
+	 */
+	protected $target_functions = array(
+		'wp_register_script' => true,
+		'wp_enqueue_script'  => true,
+		'wp_register_style'  => true,
+		'wp_enqueue_style'   => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		// Check to see if a source ($src) is specified.
+		if ( ! isset( $parameters[2] ) ) {
+			return;
+		}
+
+		/*
+		 * Version Check
+		 *
+		 * Check to make sure the version is set explicitly.
+		 */
+
+		if ( ! isset( $parameters[4] ) || $parameters[4]['raw'] === 'null' ) {
+			$this->phpcsFile->addError(
+				'Version not set for %s.',
+				$stackPtr,
+				'MissingVersion',
+				array( $matched_content )
+			);
+		} else {
+			// The version argument should have a non-false value.
+			if ( $this->is_falsy_parameter( $parameters[4] ) ) {
+				$this->phpcsFile->addError(
+					'Version ($ver) is not explicitly set for %s; Please supply an explicity version ($ver) argument',
+					$stackPtr,
+					'NoExplicitVersion',
+					array( $matched_content )
+				);
+			}
+		}
+
+		/*
+		 * In footer Check
+		 *
+		 * Check to make sure that $in_footer is set to true.
+		 * It will warn the user to make sure it is intended.
+		 *
+		 * Only wp_register_script and wp_enqueue_script need this check,
+		 * as it is not available to wp_register_style and wp_enqueue_style.
+		 */
+		switch ( $matched_content ) {
+			case 'wp_register_script':
+			case 'wp_enqueue_script':
+				if ( ! isset( $parameters[5] ) || $this->is_falsy_parameter( $parameters[5] ) ) {
+					// If in footer is not set, throw a warning about the default.
+					$this->phpcsFile->addWarning(
+						'In footer ($in_footer) is set to `false` for %s; ' .
+						'It is recommended to load scripts in the footer, please check that this is intended.',
+						$stackPtr,
+						'NotInFooter',
+						array( $matched_content )
+					);
+				}
+		}
+	}
+} // End class.
```

Comment:
We've recently removed all `// End` comments in the WPCS code. Please remove them from this PR as well.